pyparsing/core.py

```diff
@@ -1504,64 +1504,43 @@ def __mul__(self, other) -> "ParserElement":
         occurrences.  If this behavior is desired, then write
         ``expr*(None, n) + ~expr``
         """
-        if other is Ellipsis:
-            other = (0, None)
-        elif isinstance(other, tuple) and other[:1] == (Ellipsis,):
-            other = ((0,) + other[1:] + (None,))[:2]
-
-        if not isinstance(other, (int, tuple)):
-            return NotImplemented
-
         if isinstance(other, int):
-            minElements, optElements = other, 0
+            minElements, maxElements = other, other
+        elif isinstance(other, tuple):
+            minElements, maxElements = (other + (None, None))[:2]
```

Comment:
This obscure-ish expression remains; I haven't yet been able to figure out a way to rewrite this that is more readable and type-safe.

Reply:
It'd be nice to consider the `other` argument as if it were a token stream, and to handle `Ellipsis` as a special case there, consuming exactly two tokens, and with `None` emitted as a fallback if the stream is empty when reading the next token.

Reply:
No, I can't think of a neater pattern for this unpacking assignment in the code at the moment, and won't make further changes to it.